src/meili/traffic_segment_matcher.cc

```diff
@@ -329,6 +337,9 @@ std::vector<traffic_segment_t> TrafficSegmentMatcher::form_segments(const std::l
     printf("\nReported Segments:\n");*/
 
     //go over the segments and move the interpolation markers accordingly
+    int idx = 0;
```

Comment:
you could, if you wanted to, compute `idx` at any time by just doing `&segment - &segments[0]`